src/celengine/name.cpp

```diff
@@ -1,120 +1,38 @@
-#include <celutil/debug.h>
-#include "name.h"
-
-uint32_t NameDatabase::getNameCount() const
-{
-    return nameIndex.size();
-}
 
-void NameDatabase::add(const uint32_t catalogNumber, const std::string& name, bool replaceGreek)
-{
-    if (name.length() != 0)
-    {
-#ifdef DEBUG
-        uint32_t tmp;
-        if ((tmp = getCatalogNumberByName(name)) != InvalidCatalogNumber)
-            DPRINTF(2,"Duplicated name '%s' on object with catalog numbers: %d and %d\n", name.c_str(), tmp, catalogNumber);
-#endif
-        // Add the new name
-        //nameIndex.insert(NameIndex::value_type(name, catalogNumber));
-        std::string fname = ReplaceGreekLetterAbbr(name);
+#include <fmt/printf.h>
+#include "name.h"
 
-        nameIndex[fname] = catalogNumber;
-        numberIndex.insert(NumberIndex::value_type(catalogNumber, fname));
-    }
-}
-void NameDatabase::erase(const uint32_t catalogNumber)
-{
-    numberIndex.erase(catalogNumber);
-}
+using namespace std;
 
-uint32_t NameDatabase::getCatalogNumberByName(const std::string& name) const
+void Name::makeHash()
 {
-    NameIndex::const_iterator iter = nameIndex.find(name);
-
-    if (iter == nameIndex.end())
+    for (const auto &ch : *m_ptr)
     {
-        iter = nameIndex.find(ReplaceGreekLetterAbbr(name));
-        if (iter == nameIndex.end())
-            return InvalidCatalogNumber;
+        m_hash += ch;
```

Comment:
i'm afraid this hash will have 100500 collisions

Reply:
Maybe... I used hash before latest octree optimisation, to check if it will increase performance.